### PR TITLE
scippify Jump_diffusion broken

### PR DESCRIPTION
**Pull request by Harry-Rich**
- Updated jump diffusion analyser to use scipp for from xdatcar method
- Added jump diffusion coefficient calculation to diffusion.py 

- still need to add changes for n_particles within _jump_diffusion_coefficient

- mstd calculation working correctly but J_D coefficient calculation is not working correctly.

### CODE REVIEW

File: kinisi/diffusion.py [hunk 1]
```diff
@@ -28,6 +28,7 @@ def __init__(self, msd: sc.DataArray):
         self.gradient = None
         self.intercept = None
         self._diffusion_coefficient = None
+        self.jump_diffusion_coefficient = None
```
Comment (arm61) [inline kinisi/diffusion.py:31]:
```suggestion
        self._jump_diffusion_coefficient = None
```

File: kinisi/diffusion.py [hunk 2]
```diff
@@ -135,12 +137,33 @@ def diffusion(self, start_dt: sc.Variable, **kwargs):
...
+        """
+        Calculation of the diffusion coefficient. 
+
+        :param start_dt: The time at which the diffusion regime begins.
+        :param kwargs: Additional keyword arguments to pass to :py:func:`bayesian_regression`.
+        """
+
+        self.bayesian_regression(start_dt=start_dt, **kwargs)
+        # * self.msd.coords['n_samples'][0].value
+        self._jump_diffusion_coefficient = sc.to_unit(self.gradient / (2 * self.msd.coords['dimensionality'].value * 192), 'cm2/s')
```
Comment (arm61) [inline kinisi/diffusion.py:151]: Can you set this to the number of atoms just now rather than the hard-coded 192?

File: docs/source/test_scipp_jda.ipynb
Comment (arm61) [inline docs/source/test_scipp_jda.ipynb:1]: I don't think we need to actually include this file in the final merge.

File: kinisi/jump_diffusion_analyzer.py [hunk 3]
```diff
@@ -31,218 +34,69 @@ class JumpDiffusionAnalyzer(Analyzer):
...
+    
     @property
-    def mstd_std(self) -> np.ndarray:
-        """
-        :return: MSD standard deviation values for the input trajectories (a single standard deviation).
-        """
-        return self._diff.s
+    def n_atoms(self):
+        """Property to access the n_atoms."""
+        return n_atoms(self.trajectory)
```
Comment (arm61) [inline kinisi/jump_diffusion_analyzer.py:88]:
```suggestion
        return sc.scalar(self.trajectory.displacements.sizes['atom'])
```

Does something like this work?

Comment (arm61) [reply kinisi/jump_diffusion_analyzer.py:88]: This way you can delete the `n_atoms` function from the `displacments.py` module.

File: kinisi/jump_diffusion_analyzer.py [hunk 4]
```diff
@@ -31,218 +34,69 @@ class JumpDiffusionAnalyzer(Analyzer):
...
-        return np.array([self.D_J.samples, self.intercept.samples]).T
+        Calculate the jump diffusion coefficient using the mean-squared total displacement data.
+        
+        :param start_dt: The time at which the diffusion regime begins.
+        :param diffusion_params: The keyword arguements for the diffusion calculation
+            (see :py:func:`diffusion.bayesian_regression`). Optional, defaults to :py:attr:`None`.
+        """
+        if diffusion_params is None:
+            diffusion_params = {}
+        self.diff = Diffusion(msd = self.mstd, n_atoms = self.n_atoms)
```
Comment (arm61) [inline kinisi/jump_diffusion_analyzer.py:101]:
```suggestion
        self.diff = Diffusion(msd=self.mstd, n_atoms=self.n_atoms)
```

Might be easier to run things through the linter. Typically we use 

```
yapf -i kinisi/*.py
``` 

which does the linting inplace (you might need to install `yapf`).

File: kinisi/displacement.py [hunk 5]
```diff
@@ -76,9 +76,16 @@ def calculate_mstd(p: parser.Parser, progress: bool = True) -> sc.Variable:
                         coords={
                             'timestep': p.dt['timestep', :len(mstd)],
                             'n_samples': sc.array(dims=['timestep'], values=n_samples),
-                            'dimensionality': p.dimensionality
+                            'dimensionality': p.dimensionality,
                         })
 
+def n_atoms(p: parser.Parser, progress: bool = True) -> sc.Variable:
```
Comment (arm61) [inline kinisi/displacement.py:82]: See later comment.

File: kinisi/displacement.py [hunk 6]
```diff
@@ -76,9 +76,16 @@ def calculate_mstd(p: parser.Parser, progress: bool = True) -> sc.Variable:
                         coords={
                             'timestep': p.dt['timestep', :len(mstd)],
                             'n_samples': sc.array(dims=['timestep'], values=n_samples),
-                            'dimensionality': p.dimensionality
+                            'dimensionality': p.dimensionality,
```
Comment (arm61) [inline kinisi/displacement.py:79]: See comment about linting.

File: kinisi/jump_diffusion_analyzer.py (hunk 3 above)
Comment (Harry-Rich) [reply kinisi/jump_diffusion_analyzer.py:88]: Makes sense, much less clunky

**Comment by Harry-Rich** [comment]
Yes happy to merge, i guess just don't merge the test_scipp_jda.

**Pull request merged**